File: lib/model/wallet_configuration.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:flutter_tpay/model/apple_pay_configuration.dart';
import 'package:flutter_tpay/model/google_pay_configuration.dart';

part 'wallet_configuration.g.dart';

/// Class responsible for storing wallet configuration.
@JsonSerializable(explicitToJson: true)
class WalletConfiguration {
  final GooglePayConfiguration googlePay;
  final ApplePayConfiguration applePay;

  WalletConfiguration({required this.googlePay, required this.applePay});

  factory WalletConfiguration.fromJson(Map<String, dynamic> json) => _$WalletConfigurationFromJson(json);
  Map<String, dynamic> toJson() => _$WalletConfigurationToJson(this);
}

File: android/src/main/kotlin/com/tpay/model/WalletConfiguration.kt
package com.tpay.model

import com.tpay.model.googlePay.GooglePayConfiguration
import org.json.JSONObject

data class WalletConfiguration(val googlePay: GooglePayConfiguration?) {
    companion object {
        private const val GOOGLE_PAY = "googlePay"

        fun fromJson(json: JSONObject): WalletConfiguration = json.run {
            WalletConfiguration(googlePay = optJSONObject(GOOGLE_PAY)?.run(GooglePayConfiguration::fromJson))
        }
    }
}